src/ray/raylet/worker_pool.cc

```diff
@@ -305,15 +305,6 @@ Process WorkerPool::StartWorkerProcess(
       // Allocated_resource_json is only used in "shim process".
       worker_command_args.push_back("--allocated-instances-serialized-json=" +
                                     allocated_instances_serialized_json);
-    } else {
-      // The "shim process" setup worker is not needed, so do not run it.
-      // Check that the arg really is the path to the setup worker before erasing it, to
-      // prevent breaking tests that mock out the worker command args.
-      if (worker_command_args.size() >= 4 &&
-          worker_command_args[1].find(kSetupWorkerFilename) != std::string::npos) {
-        worker_command_args.erase(worker_command_args.begin() + 1,
-                                  worker_command_args.begin() + 4);
-      }
```

Comment:
I have made some simple benchmark, there is a little performance implications which is about less than 2 milliseconds. I think so that's really not important.

Reply:
@architkulkarni Sorry my mistake. My previous benchmark result is wrong. In my docker container, I found the latency increase about 400ms between worker process starting and worker registering.  I think it is acceptable.

Reply:
Per @stephanie-wang this seems to have enough perf impact to fail some of the nightly stress tests. This means it's likely user visible, so we should probably keep the optimization.